src/components/PeriodDimension/FixedPeriodFilter.js

```diff
@@ -50,6 +62,7 @@ FixedPeriodFilter.propTypes = {
     currentYear: PropTypes.string.isRequired,
     onSelectPeriodType: PropTypes.func.isRequired,
     onSelectYear: PropTypes.func.isRequired,
+    allowedPeriodTypes: PropTypes.arrayOf(PropTypes.string),
```

Comment:
One could argue that as the code disregards the filtering of periodTypes when `allowedPeriodTypes` isn't passed, the invert of `allowedPeriodTypes` should be passed ("disallowedPeriodTypes"?) instead. As if nothing is passed all are allowed. 

I.e. not passing the prop would result in `"no disallowedPeriodTypes" = "all periodTypes"` instead of the semantically incorrect case of `"no allowedPeriodTypes" = "all periodTypes"`, as one might assume that `"no allowedPeriodTypes" = "no periodTypes at all"`.

But I assume that the use-case is to pass the allowed ones, not the opposite, so then this makes sense I guess 🤷‍♂️

Reply:
I agree with the reasoning above, it would be more logical to end up with an exclude-list, i.e. `disallowedPeriodTypes`. But as you guessed, the use-case is usually to only allow a small handful of periodTypes.

There is another reason for going with the allow-list as well: In the couple of years I've worked here, I've seen a few new period-types emerge. Once a new one is added this component will need to be updated and that's fine. However, if we were to go with an exclude-list approach, all apps would have to update their `disallowedPeriodTypes` array whenever this happens. Using an allow-list approach would mean the new periodType wouldn't magically start appearing in these dropdowns after a version upgrade.

So yeah, I think sticking to the current implementation with `allowedPeriodTypes` seems best, even though the "ignore when undefined" part is a bit special.

I'll resolve this comment and will keep this part as it is.